server.py

```diff
@@ -28,71 +28,106 @@
 logger.info("START", version=__version__)
 
 
-def _get_next_sequence(seq):
-    logger.debug("Obtaining next sequence number")
-    try:
-        logger.debug("Getting DB connection")
-        result = db.engine.execute(seq.next_value())
-        logger.debug("Executing get next value on sequence")
-        sequence_no = result.first()[0]
-        logger.debug("Retrieved sequence", sequence=sequence_no)
-        return sequence_no
-    except (psycopg2.Error, SQLAlchemyError) as e:
-        logger.error("Error executing sequence", exception=str(e))
-        return abort(500)
+def sequence_values(seq, n=1):
+    logger.debug("Obtaining next {0} of sequence".format(n))
+    while n:
+        try:
+            result = db.engine.execute(seq.next_value())
+            yield result.first()[0]
+            n -= 1
+        except (psycopg2.Error, SQLAlchemyError) as e:
+            logger.error("Error executing sequence", exception=str(e))
+            return
 
 
 @app.route('/sequence', methods=['GET'])
 def do_get_sequence():
     """Get the next sequence number. Starts at 1000 and increments to 9999."""
-    sequence_no = _get_next_sequence(sequence)
-
-    # Sequence numbers start at 1000 and increment to 9999
     sequence_start = 1000
     sequence_range = 9000
+    try:
+        n = int(request.args.get("n", 1))
+    except (TypeError, ValueError):
+        return abort(400)
 
-    sequence_no = (sequence_no - 1) % sequence_range + sequence_start
+    rv = {
```

Comment:
If sequence_values experiences a pscopg2.Error or SQLAlchemyError, that entry in the list will be `None`, which isn't ideal.